### PR TITLE
Translate Scheduled Tasks

### PR DESCRIPTION
**Pull request by ferferga**
Let's see if we can slowly get rid of all the hardcoded strings and get proper localization for all languages everywhere.

I also removed a string that I discovered by chance that wasn't present anymore in the code. Not sure though if it's from a plugin? If so, I will remove that commit from the PR.

### CODE REVIEW

File: Emby.Server.Implementations/Localization/Core/en-US.json [hunk 1]
```diff
@@ -92,5 +91,22 @@
     "UserStoppedPlayingItemWithValues": "{0} has finished playing {1} on {2}",
     "ValueHasBeenAddedToLibrary": "{0} has been added to your media library",
     "ValueSpecialEpisodeName": "Special - {0}",
-    "VersionNumber": "Version {0}"
+    "VersionNumber": "Version {0}",
+    "TasksMaintenance": "Maintenance",
+    "TasksLibrary": "Library",
+    "TasksApplication": "Application",
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Localization/Core/en-US.json]: How about prefixing these three with `TaskCategory` (i.e. `TaskCategoryMaintenance`)

File: Emby.Server.Implementations/ScheduledTasks/Tasks/DeleteCacheFileTask.cs [hunk 2]
```diff
@@ -25,6 +26,7 @@ public class DeleteCacheFileTask : IScheduledTask, IConfigurableScheduledTask
         private readonly ILogger _logger;
 
         private readonly IFileSystem _fileSystem;
+        private readonly ILocalizationManager _localization;
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/DeleteCacheFileTask.cs]: This is never initialized

File: Emby.Server.Implementations/Localization/Core/en-US.json (hunk 1 above)
Comment (ferferga) [reply Emby.Server.Implementations/Localization/Core/en-US.json]: Good point

File: Emby.Server.Implementations/ScheduledTasks/Tasks/DeleteCacheFileTask.cs (hunk 2 above)
Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/DeleteCacheFileTask.cs]: Why @mark-monteiro ? Newbie in C#, I don't understand exactly why

Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/DeleteCacheFileTask.cs]: @mark-monteiro I think that I know what you mean, give me a second

**Comment by ferferga** [comment]
I made more commits than I wanted to, but many tasks were widespread all around the code. Here is an screenshot of my branch, just built:

image

File: Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs [hunk 3]
```diff
@@ -95,25 +98,18 @@ public async Task Execute(CancellationToken cancellationToken, IProgress<double>
...
 
-        /// <inheritdoc />
-        public string Description => "Downloads and installs updates for plugins that are configured to update automatically.";
+        public string Description => _localization.GetLocalizedString("TaskUpdatePluginsDescription");
 
-        /// <inheritdoc />
-        public string Category => "Application";
+        public string Category => _localization.GetLocalizedString("TasksCategoryApplication");
 
-        /// <inheritdoc />
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs]: These <inheritdoc/> tags should not be removed

File: Emby.Server.Implementations/Localization/Core/en-GB.json [hunk 4]
```diff
@@ -76,7 +76,6 @@
     "StartupEmbyServerIsLoading": "Jellyfin Server is loading. Please try again shortly.",
     "SubtitleDownloadFailureForItem": "Subtitles failed to download for {0}",
     "SubtitleDownloadFailureFromForItem": "Subtitles failed to download from {0} for {1}",
-    "SubtitlesDownloadedForItem": "Subtitles downloaded for {0}",
     "Sync": "Sync",
     "System": "System",
     "TvShows": "TV Shows",
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Localization/Core/en-GB.json]: Do the new keys need to be added here or any other languages, or do they fall back to en-US by default?

Comment (ferferga) [reply Emby.Server.Implementations/Localization/Core/en-GB.json]: @mark-monteiro Fall back to en-US afaik. I also added es because I know it, so I didn't see a reason to not do it.

File: Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs (hunk 3 above)
Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs]: Thought they were useless. Restoring them.

Comment (ferferga) [reply Emby.Server.Implementations/ScheduledTasks/Tasks/PluginUpdateTask.cs]: Thought they were useless. Restoring them.

**Comment by ferferga** [comment]
@mark-monteiro That's how it's supossed to work. It's a mess and confusing imho, but it's the current implementation. Lang for clients can be configured tapping on the user profile on the right and > Display and Lang, while server must be configured from the dashboard.

That means that I, as an admin, can have the server in Spanish but you, as an user, can see the whole interface in English if you want to. So everyone it's happy. But yeah, confusing.

File: Emby.Server.Implementations/Localization/Core/en-US.json [hunk 5]
```diff
@@ -92,5 +91,27 @@
     "UserStoppedPlayingItemWithValues": "{0} has finished playing {1} on {2}",
     "ValueHasBeenAddedToLibrary": "{0} has been added to your media library",
     "ValueSpecialEpisodeName": "Special - {0}",
-    "VersionNumber": "Version {0}"
+    "VersionNumber": "Version {0}",
+    "TasksCategoryMaintenance": "Maintenance",
+    "TasksCategoryLibrary": "Library",
+    "TasksCategoryApplication": "Application",
+    "TasksCategoryChannels": "Canales de internet",
```
[the remark this answers is not in the document]
Comment (ferferga) [reply Emby.Server.Implementations/Localization/Core/en-US.json]: Thank you very much for your patience with me, you really need it lol 🤦‍♂️🤦‍♂️

**Comment by ferferga** [comment]
@mark-monteiro Maybe I can open an issue about that so you guys have it in your roadmap?

**Comment by mark-monteiro** [comment]
> @mark-monteiro Shall I open an issue about that so you guys have it in your roadmap?

Yeah, might as well have the other back-end people review and see if its something we actually want to do. You should include a link to [this documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.1#implement-a-strategy-to-select-the-languageculture-for-each-request) in the issue.

> Everything should be fine with my last commit, sorry for all the hassle.

All good, it's just how code review goes haha.

**Comment by ferferga** [comment]
@jellyfin/backend As it seems that there will be a 10.5.3, maybe we can backport this change as well?

File: Emby.Server.Implementations/Localization/Core/en-US.json [hunk 6]
```diff
@@ -75,7 +75,6 @@
     "Songs": "Songs",
     "StartupEmbyServerIsLoading": "Jellyfin Server is loading. Please try again shortly.",
     "SubtitleDownloadFailureFromForItem": "Subtitles failed to download from {0} for {1}",
-    "SubtitlesDownloadedForItem": "Subtitles downloaded for {0}",
```
Comment (JustAMan) [inline Emby.Server.Implementations/Localization/Core/en-US.json]: why is this removed?

File: Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs [hunk 7]
```diff
@@ -18,27 +19,30 @@ public class RefreshChannelsScheduledTask : IScheduledTask, IConfigurableSchedul
...
-        public string Name => "Refresh Channels";
+        public string Name => _localization.GetLocalizedString("TasksRefreshChannels");
 
         /// <inheritdoc />
-        public string Description => "Refreshes internet channel information.";
+        public string Description => _localization.GetLocalizedString("TasksRefreshChannelsDescription");
 
         /// <inheritdoc />
-        public string Category => "Internet Channels";
+        public string Category => _localization.GetLocalizedString("TasksCategoryChannels");
```
Comment (JustAMan) [inline Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: I don't exactly like the proposed naming scheme - in `Description` the word `Description` is postfixed to the task name, and in `Category` it's prefixed.
I propose to name them (for task `Foo`) like this:
 - `Name` => `TaskFooName`
 - `Description` => `TaskFooDescription`
 - `Category` => `TaskFooCategory`

Comment (mark-monteiro) [reply Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: `TaskFooName` makes sense, but the categories aren't associated with a specific task, there are only four of them "Maintenance",  "Library", "Application", and "Internet Channels", and each one is associated with multiple tasks

File: Emby.Server.Implementations/Localization/Core/en-US.json (hunk 6 above)
Comment (ferferga) [reply Emby.Server.Implementations/Localization/Core/en-US.json]: @JustAMan Not used anywhere, simply dead. Probably there are a lot of other strings which are not used, but I found this one only while tinkering with multiple strings.

I plan on check all the dead strings deeply once I get some localization stuff sorted out in Web and everywhere else, do you think it's better to hold this one back to that future PR?

File: Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs [hunk 8]
```diff
@@ -18,27 +19,30 @@ public class RefreshChannelsScheduledTask : IScheduledTask, IConfigurableSchedul
...
             _channelManager = channelManager;
             _userManager = userManager;
             _logger = logger;
             _libraryManager = libraryManager;
+            _localization = localization;
         }
 
         /// <inheritdoc />
-        public string Name => "Refresh Channels";
+        public string Name => _localization.GetLocalizedString("TasksRefreshChannels");
```
Comment (Bond-009) [inline Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: Are you sure this is never used to ID the task?

Comment (ferferga) [reply Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: @Bond-009 That's the purpose of the ``Key`` variable imo. I didn't have any problems running the tasks in the version I built. Visually, they were progressing correctly, although I didn't look at the logs to be fair.

File: Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs (hunk 7 above)
Comment (JustAMan) [reply Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: In this case name it `TaskBarCategory` for category `Bar`. My point is, this should always be a suffix, not a mish-mash of suffix vs prefix.

Comment (ferferga) [reply Emby.Server.Implementations/Channels/RefreshChannelsScheduledTask.cs]: @JustAMan Now I see your point. Changes being pushed right now...

**Comment by ferferga** [comment]
@JustAMan As I said to @Bond-009, I didn't have any problems running the tasks in the version I built. Visually, they were progressing correctly. Didn't dive in the logs to check but the tasks were doing their thing, so I suppose that's a yes? :)
